### ISSUE TITLE
Queue priority management

### ISSUE THREAD

**Issue by faroligo** [Harxy/Jschat#106]
I propose that the queue gets prioritized by user so that users can't fill the queue with their songs and then others have to wait for ever.

Example:
Paul adds 10 items to queue
Steve adds 2 items to queue
Oli adds 2 items to queue

Queue would be: Paul 1, Steve 1, Oli 1, Paul 2, Steve 2, Oli 2, Paul 3, Paul 4..... until someone else dropped songs in.

Thoughts?

**Comment by Harxy** [comment]
think this is an excellent idea... would be good for someone joining a room
too... would cut wait time by a whole heap.

**Comment by meadsteve** [comment]
yup great idea. It's going to end up being medium complicated but still a good idea.

**Comment by meadsteve** [comment]
I'd argue this is less urgent however as we can always initially rely on an honour system
